docs/source/workflows.mlip_fitting.md

```diff
@@ -1,2 +1,190 @@
+
+```python
+import os
+os.environ['WFL_NUM_PYTHON_SUBPROCESSES'] = "4"
+os.environ['WFL_GAP_FIT_OMP_NUM_THREADS'] = "4"
+
+import wfl.autoparallelize
+wfl.autoparallelize.mpipool_support.init(verbose=4)
+```
+The "WFL_NUM_PYTHON_SUBPROCESSES" being the number of cores you provide the wfl package with, and "WFL_GAP_FIT_OMP_NUM_THREADS" being the number of cores the GAP fit can run in parallel. We initialize the parallelisation by running the [mpipool_support](overview.parallelisation.rst).
+
+Next we import all necessary functions. As you can see, for this example the only non-basic packages are quippy and ase.
+
+```python
+import json, os, yaml
+import numpy as np
+
+from ase.io import read, write
+from ase.calculators.emt import EMT
+
+from quippy.potential import Potential
+
+from wfl.calculators.generic import run as generic_calc
+from wfl.descriptors.quippy import from_any_to_Descriptor
+from wfl.descriptors.quippy import calc as desc_calc
+from wfl.configset import ConfigSet, OutputSpec
+from wfl.fit.gap.multistage import prep_params
+from wfl.fit.gap.multistage import fit as gap_fit
+from wfl.fit.ref_error import calc as ref_calc
+from wfl.generate.md import sample as sample_md
+from wfl.generate.optimize import run as optimize
+from wfl.select.by_descriptor import greedy_fps_conf_global
+```
+
+## Fitting the initial GAP
+
+The ```main``` function in ```batch_gap_fit.py``` begins with fitting an initial GAP for Cu structures:
+
+```python
+### GAP parameters
+gap_params = 'multistage_gap_params.json'
+with open(gap_params, 'r') as f:
+  gap_params = json.loads(f.read())
+Zs = [29]
+length_scales = {
+  29: {
+    "bond_len": [2.6, "NB VASP auto_length_scale"],
+    "min_bond_len": [2.2, "NB VASP auto_length_scale"],
+    "other links": {},
+    "vol_per_atom": [12, "NB VASP auto_length_scale"]
+  }
+}
+
+training = 'EMT_atoms.xyz'
+```
+The dictionary of located in ```multistage_gap_params.json``` contains all the necessary GAP hyperparameters for a [multistage fit](wfl.fit.gap.rst). As we are only investigating Cu, the list of unique atomic numbers only has one value, ```Zs=[29]```. To include SOAP heuristics, we add a dictionary of length scales specific to the Cu atom. For more information, we refer to the [universalSOAP package](https://github.com/libAtoms/universalSOAP). To generalize this part of the code for other systems, we suggest using the previously VASP-calculated dictionary located there. The training data is located in the xyz-file ```EMT_atoms.xyz```. This file can be any ase-readable type file, and includes one structure with only one Cu atom, no periodic boundary conditions, and the property ```config_type=isolated_atom```. The energy of this structure being our default Cu energy.
+
+```python
+### Initial GAP training
+fit_idx = 0
+gap_name = f'GAP_{fit_idx}'
+GAP_dir = 'GAP'
+if not os.path.isdir(GAP_dir):
+  os.mkdir(GAP_dir)
```

Comment:
If you switched to `pathlib.Path` for your file objects, you could have cleaner code like `GAP_dir.mkdir(exist_ok=True)`.  Not sure it's worth it, but I generally like `Path` over string filenames and the various `os.*` functions.